prime-angular-frontend/src/app/modules/adjudication/shared/guards/adjudication.guard.ts

```diff
@@ -33,35 +34,40 @@ export class AdjudicationGuard extends BaseGuard {
    * Check the user is authenticated, otherwise redirect
    * them to an appropriate destination.
    */
-  // TODO update to be two observables merged and resolved using combineLatest,
-  // but requires wrapping the Keycloak service so it uses obseravables first
   protected checkAccess(routePath: string = null): Observable<boolean> | Promise<boolean> {
-    const admin$ = this.authService.getAdmin$()
+
+    return this.authService.getAdmin$()
       .pipe(
         exhaustMap((admin: Admin) => {
-          // Attempt to create an admin if they don't exist
-          return (this.permissionService.hasRoles(Role.ADMIN))
-            ? this.adjudicationResource.createAdmin(admin)
-            : Promise.resolve(admin);
+          return this.adjudicationResource.getAdjudicatorByUserId(admin.userId)
+            .pipe(
+              exhaustMap((primeAdmin: Admin) => {
+                return of([primeAdmin, admin]);
+              }));
+        }),
+        exhaustMap(([primeAdmin, admin]: [Admin, Admin]) => {
+          if (primeAdmin) {
+            //if PRIME admin record found, pass the status to next step
+            admin.status = primeAdmin.status
+          } else {
+            // if PRIME does not have any admin user with the user id, create it
+            this.adjudicationResource.createAdmin(admin);
```

Comment:
@bergomi02 there is some problem with logic here as when I log into PR environment, no `Admin` row is created

Reply:
It should be fixed.